tutorials/satellite_scheduler.py

```diff
@@ -0,0 +1,272 @@
+                                           num_satellites)
+        else:
+            self.view_coords = view_coords
+        self.num_requests = num_requests
+        self.turn_rate = turn_rate
+        self.qubo_weights = qubo_weights
+        if type(solution_criteria) is float and 0.0 < solution_criteria <= 1.0:
+            self.target_cost = int(-solution_criteria * num_requests * qubo_weights[0])
+        elif type(solution_criteria) is int and solution_criteria < 0:
+            self.target_cost = solution_criteria
+        self.random_seed = None
+        self.graph = None
+        self.adjacency = None
+        self.satellites = None
+        self.requests = None
+        self.netx_solution = None
+        self.lava_backend = None
+        self.probe_cost = None
+        self.qubo_problem = None
+        self.solver_report = None
+        self.lava_solution = None
+    
+    def generate(self, seed=None):
+        """ Generate a new scheduler problem. """
+        if seed:
+            self.random_seed = seed
+            np.random.seed(seed)
+        self.graph = ntx.Graph()
+        self.satellites = range(self.num_satellites)
+        self.generate_requests()
+        self.generate_visible_nodes()
+        self.generate_infeasibility_graph()
+        self.rescale_adjacency()
+    
+    def generate_requests(self):
+        """ Generate a random set of requests in the 2D plane. """
+        self.requests = np.random.random((self.num_requests, 2))
+        order = np.argsort(self.requests[:,0])
+        self.requests = self.requests[order,:]
+
+    def generate_visible_nodes(self):
+        """
+        Add nodes to the graph for every combination of a vehicle and a request visible
+        to that vehicle.
+        """
+        node_id = 0
+        for i in self.satellites:
+            for j in range(self.num_requests):
+                if is_visible(self.view_coords[i], self.requests[j,1], self.view_height):
+                    self.graph.add_node((node_id, i, self.requests[j,0], self.requests[j,1]))
+                    node_id += 1
+        self.num_nodes = node_id
+    
+    def generate_infeasibility_graph(self):
+        """
+        Create edges between any two nodes of the graph which cannot be traversed and
+        add corresponding weights to the adjacency matrix.
+        """
+        self.adjacency = np.zeros((self.num_nodes, self.num_nodes), dtype=int)
+        for n1 in self.graph.nodes:
+            for n2 in self.graph.nodes:
+                if not is_same_node(n1, n2) and not is_feasible(n1, n2, self.turn_rate):
+                    self.adjacency[n1[0], n2[0]] = 1
+                    self.graph.add_edge(n1, n2)
+    
+    def rescale_adjacency(self):
+        """ Scale the adjacency matrix weights for QUBO solver. """
+        self.adjacency = np.triu(self.adjacency)
+        self.adjacency += self.adjacency.T - 2 * np.diag(self.adjacency.diagonal())
+
+    def solve_with_netx(self):
```

Comment:
To have an interesting comparison, I would probably substitute the approximate solver with the exact one from netx (the same we use in MISProblem).

Reply:
I'll look at this again. When I was initially setting this up, the netx solution used in MIS was not working for me, but I might have been doing something wrong.

Reply:
Okay, after looking at this again, I think it's reasonable, but it will require a bit more code than I can write at this moment. Since it's not a functional blocker, I'd prefer to open an issue. Hope that works for you.